### PR TITLE
Updated flux version and configs

### PR DESCRIPTION
**Pull request by cartyc**
Updated the flux image to version `1.17.0` and memcached to `1.5.20`. I also updated the flux deployment to be more inline with what is in the main [repo](https://github.com/fluxcd/flux/tree/master/deploy) .

I tested the changes locally and they seem to work and ran kubeval against the new output.

```
▶ k kustomize install | kubeval -
PASS - stdin contains a valid Namespace
PASS - stdin contains a valid ServiceAccount
PASS - stdin contains a valid Role
PASS - stdin contains a valid ClusterRole
PASS - stdin contains a valid ClusterRole
PASS - stdin contains a valid ClusterRole
PASS - stdin contains a valid RoleBinding
PASS - stdin contains a valid ClusterRoleBinding
PASS - stdin contains a valid ClusterRoleBinding
PASS - stdin contains a valid ConfigMap
PASS - stdin contains a valid Secret
PASS - stdin contains a valid Service
PASS - stdin contains a valid Deployment
PASS - stdin contains a valid Deployment
PASS - stdin contains a valid PodSecurityPolicy
```

Thanks!

### CODE REVIEW

File: base/flux/deployment.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ spec:
   template:
     metadata:
       annotations:
-        prometheus.io.scrape: "true"
+        prometheus.io/port: "3031"
```
Comment (stefanprodan) [inline base/flux/deployment.yaml:14]: You need to set this port in Flux args with `--listen-metrics=:3031` and please enable scraping.

**Review by stefanprodan — CHANGES_REQUESTED** [review]

File: base/flux/deployment.yaml (hunk 1 above)
Comment (cartyc) [reply base/flux/deployment.yaml:14]: Added the `--listen-flag` and renabled `prometheus.io.scrape` . Out of curiosity is there a reason why scrape is not enabled in the main [Flux deployment] (https://github.com/fluxcd/flux/blob/master/deploy/flux-deployment.yaml)?

Comment (stefanprodan) [reply base/flux/deployment.yaml:14]: For consistency please rename it to `prometheus.io/scrape`. Thanks

Comment (cartyc) [reply base/flux/deployment.yaml:14]: Just pushed the fixed annotation :)

**Review by stefanprodan — APPROVED** [review]

**Pull request merged**